### PR TITLE
FLUME-1342 adding jmx metrics tables to docs

### PR DESCRIPTION
**Pull request by majorendre**
This PR adds a few tables to the User Guide that describe the metrics published by sorurces, sinks and channels.
I used simple unix tools to gather the data then I wrote a small utility to convert it to csv.
Then I used an online converter https://www.tablesgenerator.com/ to generate the rst tables and then a little manual editing.
I discovered some rst formatting problems in the FlumeUserGuide.rst, corrected them, too.
It was rather painful process to gather the data and find a decent representation.
So far this PR only contains the end result. I would be happy to share the utilities, just don't know what would be the best way.

### CODE REVIEW

**Comment by adenes** [comment]
Thank you @majorendre , this seems to be a really useful addition to our documentation.
LGTM, the `FlumeUserGuide.html` gets rendered correctly including the newly added tables, but I haven't checked their content.

File: flume-ng-doc/sphinx/FlumeUserGuide.rst [hunk 1]
```diff
@@ -4944,6 +4942,176 @@ Monitoring in Flume is still a work in progress. Changes can happen very often.
...
+| KafkaEventGetTimer       |                   |                |           |           |         |        |
++--------------------------+-------------------+----------------+-----------+-----------+---------+--------+
+| OpenConnectionCount      |                   |                |           |           |         |        |
++--------------------------+-------------------+----------------+-----------+-----------+---------+--------+
+
+Sinks 1
+~~~~~~~
+
++------------------------+-------------+------------+--------------+---------------+-------+--------+
+|                        | AbstractRpc | AsyncHBase | BucketWriter | ElasticSearch | HBase | HBase2 |
```
Comment (szaboferee) [inline flume-ng-doc/sphinx/FlumeUserGuide.rst]: I guess the BucketWriter is the HDFS sink and the AbstractRpc means some other implementation that will be in the metrics. Could you please check it?

**Review by szaboferee — COMMENTED** [review]
Thanks for the contribution @majorendre 
I have one question below, other than that it looks good

File: flume-ng-doc/sphinx/FlumeUserGuide.rst (hunk 1 above)
Comment (majorendre) [reply flume-ng-doc/sphinx/FlumeUserGuide.rst]: Thanks for spotting. Corrected them.

**Pull request closed without merging**